uml4net.xmireader/Readers/XmiReader.cs

```diff
@@ -0,0 +1,234 @@
+        /// The <see cref="IAssembler"/>
+        /// </summary>
+        private readonly IAssembler assembler;
+
+        /// <summary>
+        /// The <see cref="ILogger"/> used to log
+        /// </summary>
+        private readonly ILogger<XmiReader> logger;
+
+        /// <summary>
+        /// The <see cref="IXmiElementReader{T}"/>of <see cref="IPackage"/>
+        /// </summary>
+        private readonly IXmiElementReader<IPackage> packageReader;
+
+        /// <summary>
+        /// The <see cref="IXmiElementReader{T}"/>of <see cref="IModel"/>
+        /// </summary>
+        private readonly IXmiElementReader<IModel> modelReader;
+
+        /// <summary>
+        /// The <see cref="IExternalReferenceResolver"/>
+        /// </summary>
+        private readonly IExternalReferenceResolver externalReferenceResolver;
+
+        /// <summary>
+        /// The <see cref="IXmiReaderScope"/>
+        /// </summary>
+        private readonly IXmiReaderScope scope;
+
+        /// <summary>
+        /// The <see cref="IXmiReaderCache"/>
+        /// </summary>
+        private readonly IXmiReaderCache cache;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmiReader"/> class.
+        /// </summary>
+        /// <param name="assembler">The <see cref="IAssembler"/></param>
+        /// <param name="cache">The <see cref="IXmiReaderCache"/></param>
+        /// <param name="logger">
+        /// The (injected) <see cref="ILogger{T}"/> used to setup logging
+        /// </param>
+        /// <param name="packageReader">The <see cref="IXmiElementReader{T}"/>of <see cref="IPackage"/></param>
+        /// <param name="modelReader">The <see cref="IXmiElementReader{T}"/>of <see cref="IModel"/></param>
+        /// <param name="externalReferenceResolver">The <see cref="IExternalReferenceResolver"/></param>
+        /// <param name="scope">The <see cref="IXmiReaderScope"/></param>
+        public XmiReader(IAssembler assembler, IXmiReaderCache cache, ILogger<XmiReader> logger, IXmiElementReader<IPackage> packageReader,
+            IXmiElementReader<IModel> modelReader, IExternalReferenceResolver externalReferenceResolver, IXmiReaderScope scope)
+        {
+            this.assembler = assembler;
+            this.cache = cache;
+            this.logger = logger;
+            this.packageReader = packageReader;
+            this.modelReader = modelReader;
+            this.externalReferenceResolver = externalReferenceResolver;
+            this.scope = scope;
+        }
+
+        /// <summary>
+        /// Reads the content of a UML XMI 2.5.1 file asynchronously.
+        /// </summary>
+        /// <param name="fileUri">
+        /// The URI of the XMI file to be read.
+        /// </param>
+        /// <returns>
+        /// An <see cref="IEnumerable{IPackage}"/> representing the deserialized packages from the XMI file.
+        /// </returns>
+        public async Task<IEnumerable<IPackage>> ReadAsync(string fileUri)
```

Comment:
is this `async Task` because we are supporting fetching a resource from the internet or networked resource using http or https? i don't think we should do that. We cannot trust the online resource.